types/sinon/index.d.ts

```diff
@@ -19,13 +19,17 @@ interface Event {} // tslint:disable-line no-empty-interface
 interface Document {} // tslint:disable-line no-empty-interface
 
 declare namespace Sinon {
-    type DeepPartialOrMatcher<T> = {
-        [K in keyof T]?: SinonMatcher | (T[K] extends object ? DeepPartialOrMatcher<T[K]> : T[K]);
+    type MatchProvidedArguments<T> = {
+        [K in keyof T]?: SinonMatcher | (T[K] extends object ? MatchProvidedArguments<T[K]> : T[K]);
     };
+    // Alias for backward compatibility, remove on next major release
+    type DeepPartialOrMatcher<T> = MatchProvidedArguments<T>;
 
-    type MatchArguments<T> = {
-        [K in keyof T]: SinonMatcher | (T[K] extends object ? MatchArguments<T[K]> : never) | T[K];
+    type MatchExactlyArguments<T> = {
+        [K in keyof T]: SinonMatcher | (T[K] extends object ? MatchExactlyArguments<T[K]> : T[K]);
     };
+    // Alias for backward compatibility, remove on next major release
```

Comment:
to make these easier to find, can you prefix with `TODO: `?

Reply:
Done